### PR TITLE
Fix some broken links, some typos, and make the sample code run.

### PR DESCRIPTION
**Pull request by clementlee**
The whole example now executes. Hope the code changes are correct! (New to Julia, so still learning here)

### CODE REVIEW

File: docs/src/contributing/adding_algorithms.md [hunk 1]
```diff
@@ -58,15 +58,20 @@ For more details, refer to https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull
 
 ```julia
 using OrdinaryDiffEq
-import OrdinaryDiffEq: OrdinaryDiffEqAlgorithm,OrdinaryDiffEqConstantCache,
-      alg_order, alg_cache, initialize!, perform_step!, @muladd, @unpack,
-      constvalue, @cache
+import OrdinaryDiffEq: 
+      OrdinaryDiffEqAlgorithm, OrdinaryDiffEqMutableCache, OrdinaryDiffEqConstantCache,
+      alg_order, alg_cache, initialize!, perform_step!, trivial_limiter!, constvalue,
+      @muladd, @unpack, @..
```
Comment (clementlee) [inline docs/src/contributing/adding_algorithms.md:64]: side note: the `@..` macro is impossible to google.

File: docs/src/contributing/adding_algorithms.md [hunk 2]
```diff
@@ -58,15 +58,20 @@ For more details, refer to https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull
...
-struct RK_ALG <: OrdinaryDiffEq.OrdinaryDiffEqAlgorithm end
+struct RK_ALG{StageLimiter,StepLimiter} <: OrdinaryDiffEq.OrdinaryDiffEqAlgorithm 
+  stage_limiter!::StageLimiter
+  step_limiter!::StepLimiter
+end
+RK_ALG(stage_limiter! = trivial_limiter!) = RK_ALG(stage_limiter!, trivial_limiter!)
 export RK_ALG
 alg_order(alg::RK_ALG) = 3
 
-@cache struct RK_ALGCache{uType,rateType,StageLimiter,StepLimiter,TabType} <: OrdinaryDiffEqMutableCache
```
Comment (clementlee) [inline docs/src/contributing/adding_algorithms.md:69]: not sure how you could cache a struct?

File: docs/src/contributing/adding_algorithms.md [hunk 3]
```diff
@@ -188,28 +193,28 @@ end
 
   # u1 -> stored as u
   @.. u = uprev + β10 * dt * integrator.fsalfirst
-  stage_limiter!(u, f, t+c1*dt)
+  stage_limiter!(u, f, p, t+c1*dt)
```
Comment (clementlee) [inline docs/src/contributing/adding_algorithms.md:196]: I'm guessing this is the right call, from reading https://github.com/SciML/OrdinaryDiffEq.jl/blob/b563fcce03ee83d354c957a8a9c4f71bb85a56c1/src/perform_step/ssprk_perform_step.jl#L43

Comment (ChrisRackauckas) [reply docs/src/contributing/adding_algorithms.md:196]: yes

File: docs/src/contributing/adding_algorithms.md (hunk 2 above)
Comment (ChrisRackauckas) [reply docs/src/contributing/adding_algorithms.md:69]: no `@cache` is right here. It's not about caching it, it's about defining the cache functions.

File: docs/src/contributing/adding_algorithms.md (hunk 1 above)
Comment (ChrisRackauckas) [reply docs/src/contributing/adding_algorithms.md:64]: FastBroadcast.jl

File: docs/src/contributing/adding_algorithms.md [hunk 4]
```diff
@@ -58,15 +58,20 @@ For more details, refer to https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull
...
+struct RK_ALG{StageLimiter,StepLimiter} <: OrdinaryDiffEq.OrdinaryDiffEqAlgorithm 
+  stage_limiter!::StageLimiter
+  step_limiter!::StepLimiter
+end
+RK_ALG(stage_limiter! = trivial_limiter!) = RK_ALG(stage_limiter!, trivial_limiter!)
 export RK_ALG
 alg_order(alg::RK_ALG) = 3
 
-@cache struct RK_ALGCache{uType,rateType,StageLimiter,StepLimiter,TabType} <: OrdinaryDiffEqMutableCache
+struct RK_ALGCache{uType,rateType,StageLimiter,StepLimiter,TabType} <: OrdinaryDiffEqMutableCache
```
Comment (ChrisRackauckas) [inline docs/src/contributing/adding_algorithms.md:74]:
```suggestion
@cache struct RK_ALGCache{uType,rateType,StageLimiter,StepLimiter,TabType} <: OrdinaryDiffEqMutableCache
```

File: docs/src/contributing/adding_algorithms.md [hunk 5]
```diff
@@ -58,11 +58,16 @@ For more details, refer to https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull
 
 ```julia
 using OrdinaryDiffEq
-import OrdinaryDiffEq: OrdinaryDiffEqAlgorithm,OrdinaryDiffEqConstantCache,
-      alg_order, alg_cache, initialize!, perform_step!, @muladd, @unpack,
-      constvalue, @cache
+import OrdinaryDiffEq: 
+      OrdinaryDiffEqAlgorithm, OrdinaryDiffEqMutableCache, OrdinaryDiffEqConstantCache,
+      alg_order, alg_cache, initialize!, perform_step!, trivial_limiter!, constvalue,
+      @muladd, @unpack, @..
```
Comment (ChrisRackauckas) [inline docs/src/contributing/adding_algorithms.md:64]:
```suggestion
      @muladd, @unpack, @..
```
```suggestion
      @muladd, @unpack, @cache, @..
```

File: docs/src/contributing/adding_algorithms.md (hunk 2 above)
Comment (clementlee) [reply docs/src/contributing/adding_algorithms.md]: @ChrisRackauckas hm ok, but that makes it not run:

```Julia
julia> using OrdinaryDiffEq
julia> import OrdinaryDiffEq:@cache, OrdinaryDiffEqMutableCache

julia> @cache struct RK_ALGCache{uType,rateType,StageLimiter,StepLimiter,TabType} <: OrdinaryDiffEqMutableCache
          u::uType
          uprev::uType
          k::rateType
         tmp::uType
         u₂::uType
         fsalfirst::rateType
         stage_limiter!::StageLimiter
         step_limiter!::StepLimiter
         tab::TabType
       end
ERROR: UndefVarError: RK_ALGCache not defined
Stacktrace:
 [1] top-level scope
   @ ~/.julia/packages/OrdinaryDiffEq/zbCbk/src/misc_utils.jl:36
```

Not sure what's up?

Comment (ChrisRackauckas) [reply docs/src/contributing/adding_algorithms.md]: `@cache` may require being in OrdinaryDiffEq.jl? Stick it in the package and see if it works then.

**Pull request merged**